### ISSUE TITLE
[Question] Did the next() method become unavailable in the Timer?

### ISSUE THREAD

**Issue by mag-chang** [Microsoft/BotBuilder#3349]
## System Information
* SDK Language: Node.js
* SDK Version: botbuilder@3.9.1
* Development Environment: localhost

## Issue Description
In the old version of SDK, the next() method worked in the setTimeout({ }) method,
but it does not work after upgrading the SDK.
Specifically, it updated from version 3.7.0 to 3.9.1.

## Code Example
```javascript
setTimeout((address) => {
    next({ response: "none", address: address })
}, 2000, session.message.address)
```

**Comment by stevengum97** [comment]
@mag-chang, What are you attempting to do with the snippet you provided? Could you provide additional code and background on the scenario? Also, which version of Node are you using?

**Comment by mag-chang** [comment]
@stevengum97,  Thanks reply. This code hopes to jump to the next Dialog if the user chooses nothing after the set time has elapsed after presenting the choices to the user.
I using version of Node.js v4.2.6.
I will post a more specific code.

```javascript
var card_list = [];
var card = new builder.HeroCard(session)
    .title('title')
    .subtitle('sub_title')
    .buttons([
        builder.CardAction.openUrl(session, 'http://example.com', 'open url'),
        builder.CardAction.imBack(session, 'select:back1', 'back1'),
        builder.CardAction.imBack(session, 'select:back2', 'back2')
    ]);

card_list.push(card);


var last_card = new builder.HeroCard(session)
    .title('last_title')
    .subtitle('')
    .buttons([
        builder.CardAction.imBack(session, 'select:back1', 'back1'),
        builder.CardAction.imBack(session, 'select:back2', 'back2'),
        builder.CardAction.imBack(session, 'select:back3', 'back3')
    ]);

card_list.push(last_card);


var msg = new builder.Message(session)
    .text('text')
    .textFormat(builder.TextFormat.xml)
    .attachmentLayout(builder.AttachmentLayout.carousel)
    .attachments(card_list);

// After the set time has passed, it jumps to the next Dialog.
setTimeout((address) => {
    next({ response: "none", address: address })
}, 2000, session.message.address)

builder.Prompts.text(session, msg);
```

**Comment by stevengum97** [comment]
@mag-chang are you getting any specific error messages when running this code, or does it simply not move to the next step and waits for the user's input?

**Comment by mag-chang** [comment]
@stevengum97 
It does simply not move to the next step and waits for the user's input.
When outputting the console.log() in the timer, it seems that the timer is operating because the log was output correctly.

**Comment by stevengum97** [comment]
I think what may have happened is that your code broke with the new 3.8.x Prompts system. The new system allows for greater customization but also creates the Prompts' dialogs differently than older versions. You may need to create a custom prompt to achieve the result you're seeking.

**Comment by mag-chang** [comment]
OK. Thank you for answering.
I will use Custom Prompt or consider implementing another method.

**Comment by stevengum97** [comment]
@mag-chang, were you able to implement a custom prompt to your satisfaction?

**Comment by mag-chang** [comment]
@stevengum97 

I tried various methods and found out something.
In the processing started by Timer, it seems that the Dialog structure in session is gone.

So, when I tried implementing to move Dialog with UniversalBot rather than session, I approached ideal processing.

Specifically, it is the following code.
```javascript
setTimeout((address) => {
    bot.beginDialog(address, 'next_dialog');
}, 2000, session.message.address)
```

I think that if you repeat trial and error several times, Ideal processing can be implemented.
Thank you.
